KAFKA-17683: Remove Zookeeper from READMEs in 4.0

mingyen066 opened the pull request:
Remove all Zookeeper related word from READMEs in 4.0
There are three READMEs that mention zookeeper:
* `README.md`
* `trogdor/README.md`
  * I ran first few commands to make sure trogdor works in kraft mode
* `vagrant/README.md`
  * There are some code related to zookeeper in the folder vagrant. Maybe we can create another ticket to see whether we need to fix it or remove it.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

mingyen066 committed:
Remove Zookeeper from READMEmd

mingyen066 committed:
Remove Zookeeper from trogdor README

mingyen066 commented:
Not sure If zookeeper info in `vagrant/REAMDME.md` also need to be removed.

chia7712 commented on trogdor/README.md:
```diff
@@ -6,13 +6,8 @@ Trogdor can run benchmarks and other workloads.  Trogdor can also inject faults
 Quickstart
 =========================================================
-First, we want to start a single-node Kafka cluster with a ZooKeeper and a broker.
+First, we want to start a single-node Kafka cluster with one broker.
```
We need to update this documentation to reflect KRaft mode. Perhaps you can add a reference to [this section](https://github.com/apache/kafka/blob/trunk/README.md?plain=1#L93) of the README.

urbandan committed:
KAFKA-17534: Add configuration to disable the heartbeats topic replication (#17413)

Introducing heartbeats.replication.enabled to explicitly disable the default heartbeats topic replication.
This change implements KIP-1089.

Reviewers: Viktor Somogyi-Vass <viktorsomogyi@gmail.com>

dajac committed:
KAFKA-17383; Update upgrade notes about removal of `offsets.commit.required.acks` (#17446)

This patch updates the upgrades note about the removal of `offsets.commit.required.acks` configuration in 4.0.

Reviewers: Lianet Magrans <lmagrans@confluent.io>

mingyen066 committed:
Merge remote-tracking branch 'upstream/trunk' into KAFKA-17683-Remove-ZK-from-Kafka-READMEs-in-4.0

mingyen066 committed:
mention Kraft in trogdor/README.md

mingyen066 commented:
> Not sure If zookeeper info in `vagrant/REAMDME.md` also need to be removed.

@chia7712 
There are some code related to zookeeper in the folder vagrant. Maybe we can create another ticket to see whether we need to fix it or remove it.

chia7712 commented:
> There are some code related to zookeeper in the folder vagrant. Maybe we can create another ticket to see whether we need to fix it or remove it.

yes, https://issues.apache.org/jira/browse/KAFKA-17815
